test/string_view-test/string_view-test.cpp

```diff
@@ -0,0 +1,349 @@
+
+	//assignment and data test
+	string_view sv_4 = str_2;
+	is_true(sv_4.data() == str_2, "sv_4.data()");
+	
+	is_true(sv_4[0] == str_2[0], "sv_4[0]");
+	is_true(sv_4[3] == str_2[3], "sv_4[3]");
+	is_true(sv_4[6] == str_2[6], "sv_4[6]");
+	is_true(sv_4[9] == str_2[9], "sv_4[9]");
+	is_true(sv_4[11] == str_2[11], "sv_4[11]");
+
+	is_true(sv_4.at(0) == str_2[0], "sv_4.at(0)");
+	is_true(sv_4.at(3) == str_2[3], "sv_4.at(3)");
+	is_true(sv_4.at(6) == str_2[6], "sv_4.at(6)");
+	is_true(sv_4.at(9) == str_2[9], "sv_4.at(9)");
+	is_true(sv_4.at(11) == str_2[11], "sv_4.at(11)");
+
+	is_true(sv_4.front() == str_2[0], "sv_4.front()");
+	is_true(sv_4.back() == str_2[sv_4.size() - 1], "sv_4.back()");
+
+
+	//5. Iterator support
+	//forward iterators
+	char sExpected[]{ "support" };
+	string_view sv_iter(sExpected);
+
+	bool iteratorTest{ true };
+	size_t size = sizeof(sExpected) - 1;
+	auto it = sv_iter.begin(), endIt = sv_iter.end();
+	size_t i;
+	for (i = 0; i < size && iteratorTest; ++i, ++it)
+		iteratorTest = (it != endIt && *it == sExpected[i]);
+
+	is_true(iteratorTest, "forward iterator order and content");
+	is_true(it == endIt, "forward iterator termination");
+
+	iteratorTest = true;
+	it = sv_iter.cbegin();
+	endIt = sv_iter.cend();
+	for (i = 0; i < size && iteratorTest; ++i, ++it)
+		iteratorTest = (it != endIt && *it == sExpected[i]);
+
+	is_true(iteratorTest, "forward const iterator order and content");
+	is_true(it == endIt, "forward const iterator termination");
+
+	//reverse iterators
+	char svrExpected[]{ "123456789" };
+	string_view sv_reverse("987654321");
+
+	iteratorTest = true;
+	size = sizeof(svrExpected) - 1;
+	auto rit = sv_reverse.rbegin(), rendIt = sv_reverse.rend();
+	for (i = 0; i < size && iteratorTest; ++i, ++rit)
+		iteratorTest = (rit != rendIt && *rit == svrExpected[i]);
+
+	is_true(iteratorTest, "reverse iterator order and content");
+	is_true(rit == rendIt, "reverse iterator termination");
+
+	iteratorTest = true;
+	rit = sv_reverse.crbegin();
+	rendIt = sv_reverse.crend();
+	for (i = 0; i < size && iteratorTest; ++i, ++rit)
+		iteratorTest = (rit != rendIt && *rit == svrExpected[i]);
+
+	is_true(iteratorTest, "reverse iterator order and content");
+	is_true(rit == rendIt, "reverse iterator termination");
+
+	//iterator on empty string_view: the loop body should not execute
+	iteratorTest = true;
+	for (const auto emp : empty_sv)
+		iteratorTest = false;
+	is_true(iteratorTest, "iterator on empty array");
+
+
+	//6. Modifiers
+	const char* sv_data{ sv.data() };
+	size_t sv_size{ sv.size() };
+	size_t n{ 1 };
+
+	sv.remove_prefix(n);
+	sv_size -= n;
+	is_true(sv.data() == sv_data + 1, "sv.remove_prefix().data()");
+	is_true(sv.size() == sv_size, "sv.remove_prefix().size()");
+	
+	sv_data = sv.data();
+
+	sv.remove_suffix(n);
+	sv_size -= n;
+	is_true(sv.data() == sv_data, "sv.remove_suffix().data()");
+	is_true(sv.size() == sv_size, "sv.remove_suffix().size()");
```

Comment:
Simplify as follows. Also, the code looks a bit simpler and a bit more naturally sequenced if we test remove_suffix first,

```suggestion
	data_expected = sv.data();
	size_expected--;
	sv.remove_suffix(n);
	is_true(sv.data() == data_expected, "sv.remove_suffix().data()");
	is_true(sv.size() == size_expected, "sv.remove_suffix().size()");
```